datamover/pkg/drivers/https/migration.go

```diff
@@ -653,3 +673,14 @@ func runjob(in *pb.RunJobRequest) error {
 
 	return ret
 }
+
+// To calculate Progress of migration process
+func progress(job *flowtype.Job, size int64, wt float64) {
+	// Migrated Capacity = Old_migrated capacity + WT(Process)*Size of Object/100
+	MigratedCapacity := job.MigratedCapacity + float64(size)*(wt/100)
+	job.MigratedCapacity = math.Round(MigratedCapacity*100) / 100
+	// Progress = Migrated Capacity*100/ Total Capacity
+	job.Progress = int64(job.MigratedCapacity * 100 / float64(job.TotalCapacity))
+	logger.Printf("[INFO] Progress %d", job.Progress)
+	db.DbAdapter.UpdateJob(job)
+}
```

Comment:
progress is used by multiple go-routines running in parallel, how to make sure it is concurrent security?

Reply:
I have tested this for multiple scenarios( in which go-routines were running in parallel ) , it has worked fine in all cases. Is there any way using which I can check for concurrent security.
Thanks

Reply:
Did you test several objects in a single migrating task, or several migrating tasks running in parallel? IMO, in theory, the concurrent security only exist in the first senario, and the problem is random, maybe you can test with more objects in a single migrating task and check if the progress is increasing all the time and finished with 100% at last.

Reply:
Yes we have tested both scenario, 1. Single migrating task with several objects and Multiple jobs running in parallel. In both scenario the progress was increasing all the time during migration and it finished with 100% at the end of migration.

Reply:
Hi, I did a test and found something wrong, details is as below, please check.
# Branch lick2Cloud/multi-cloud > multi_cloud_feature
## Source bucket
Totally 42 objects with the same size, as below:
![source bucket1](https://user-images.githubusercontent.com/38932432/64599780-93631e80-d3ec-11e9-97bf-9cec825b35da.png)
![source bucket2](https://user-images.githubusercontent.com/38932432/64599824-ae359300-d3ec-11e9-99f0-e4052e062aed.png)
## Result
** What get from the dashboard
![dashboard-1](https://user-images.githubusercontent.com/38932432/64600009-010f4a80-d3ed-11e9-81f7-035d38b2e1e5.png)
** What get from database
![db-1](https://user-images.githubusercontent.com/38932432/64600057-11272a00-d3ed-11e9-858c-7d29c59c7a57.png)
** What get from log
![log-1](https://user-images.githubusercontent.com/38932432/64600078-1edcaf80-d3ed-11e9-9046-32e44fcf8715.png)

# Branch opensds/multi-cloud > master
## Source bucket
It's the same as above.
## Result:
** What get from the dashboard
![dashboard-2](https://user-images.githubusercontent.com/38932432/64600228-56e3f280-d3ed-11e9-8e1c-b873d6092f38.png)
** What get from database
![db-2](https://user-images.githubusercontent.com/38932432/64600250-6105f100-d3ed-11e9-9211-1955199b45fe.png)
** What get from log
![log-2](https://user-images.githubusercontent.com/38932432/64600280-6ebb7680-d3ed-11e9-97ee-c21a78c2c349.png)

Reply:
@sfzeng Thanks for your time and testing this feature. By mistake `Passedcapacity` got deleted while adding commits. Now I have solved this issue. I am testing it again. Thanks